### DISCUSSION TITLE
Output on complex terrain with cavities

### DISCUSSION THREAD

**Discussion by mpachera** [firemodels/fds#12070] [Q&A] [answered]
Hi,

I am working on a tunnel project with a complext slope that can't be represented using the tilted gravity vector. Therefore, I modelled the whole geometry and ran the simulation. I would like to monitor some quantities at a fixed height over the tunnel floor so I thought I could use the following line to plot the results (&SLCF AGL_SLICE=1., QUANTITY='VELOCITY' /). However, when I check the results of a simplified case, these look a bit weird.
![Capture](https://github.com/firemodels/fds/assets/15084725/086c0f96-006c-4c88-8b34-f29280af9a8c)
I have some alternatives like placing devices all over the floor at variable heights or using 3D slcf, however, the output is not visually as nice as the AGL_SLICE. 

Is there a way to exclude some OBST or VENT from the AGL_SLICE so I could plot the results only inside the tunnel and exclude the plot above the roof?

I know this output was developed for terrain analyses, but this feature could be very useful for complex problems in general. 

Thanks in advance,

Matteo

**Comment by mcgratta** [comment]
Glenn -- if one uses a 3D slice file, can an image like this be created?

**Comment by gforney** [comment]
If I understand correctly what you would like to do, yes.

1. load your 3d slice file
2. open up the slice motion dialog box using the  menu entry: 
 Dialogs>Motion/View/Render then click on Motion and then Slice Motion (see attached dialog box)
4. click on show data to show plane (you can also click on the w key to show/hide the arbirtray slice plane)
manipulate rotation center and normal direction dialog entries to position the plane where you want

if you double click on othe slice plane you can rotate the slice plane
if you double click while pressing the ALT key you move the slice plane up and down.
I find this is ok to get you plane roughly where you want but the dialog box is better to get more precision
![slice_plane](https://github.com/firemodels/fds/assets/12403014/cd588151-aa82-4912-ad27-d9a0a97afd13)

**Comment by mpachera** [comment]
Hi Glenn, 

Thanks for your reply, I think I didn't present my case clearly. With 3D slicefile I can rotate my plane in space, however, I'm still forced to plot results on a plane. In my real case, the curvature varies in X and Y and the slope is not constant so if I use a slice plane I will be 2 m above the floor in some points, but a bit further I will be at lower or higher heights because my slope is not constant. That's why I tried to use the AGL_SLICE to follow the terrain.

**Comment by gforney** [comment, marked as answer]
unfortunately, at present the closest we can come to visualizing a slice the way you would like is to use a 3D slice positioned some distance above your OBSTs.  fds does not distinguish between OBSTs used for terrain and OBSTs used for buildings on the terrain (as far as I know) so an AGL slice will be shown above any buildings on the terrain as if they were terrain.
